### DISCUSSION TITLE
Checking type size

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#1546] [Q&A]
xmake中如何测试type占用的字节数？类似cmake中的checktypesize：

```cmake
include(CheckTypeSize)
CHECK_TYPE_SIZE(long SIZEOF_LONG)
CHECK_TYPE_SIZE(off_t SIZEOF_OFF_T)

if ("${SIZEOF_OFF_T}" LESS 8)
  option(CGNS_ENABLE_LFS "Enable or disable large file support" "OFF")
endif ("${SIZEOF_OFF_T}" LESS 8)
```

**Comment by waruqi** [comment]
目前只能snippet检测，走 `includes("check_cxsnippets.lua")` 里面的辅助接口，check_csnippets/check_cxxsnippets

```lua
check_cxxsnippets("CGNS_ENABLE_LFS", [[#include <stdlib.h>
static_assert(sizeof(off_t) == 8, "");]])
```

或者直接走 option 里面的 add_cxxsnippets/add_csnippets

**Reply by jingkaimori** [reply]
有些autoconf导出的包，必须把类型占用的字节数写到配置文件里：
```
/* The size of `int', as computed by sizeof. */
#undef SIZEOF_INT

/* The size of `intmax_t', as computed by sizeof. */
#undef SIZEOF_INTMAX_T

/* The size of `intptr_t', as computed by sizeof. */
#undef SIZEOF_INTPTR_T
```

**Comment by waruqi** [comment]
> 有些autoconf导出的包，必须把类型占用的字节数写到配置文件里：
> 
> ```
> /* The size of `int', as computed by sizeof. */
> #undef SIZEOF_INT
> 
> /* The size of `intmax_t', as computed by sizeof. */
> #undef SIZEOF_INTMAX_T
> 
> /* The size of `intptr_t', as computed by sizeof. */
> #undef SIZEOF_INTPTR_T
> ```

我记得是支持了，捕获输出 output = true

https://github.com/xmake-io/xmake/blob/68153d06e61342aa2be927116c9b7d4e40125945/tests/apis/check_xxx/xmake.lua#L46
